controllers/account/region_enablement.go

```diff
@@ -0,0 +1,349 @@
+	currentInFlightCount, inFlightOptInRequests := currentAcctInstance.GetOptInRequestsByStatus(awsv1alpha1.OptInRequestEnabling)
+	_, onlyOpenOptInRequests := currentAcctInstance.GetOptInRequestsByStatus(awsv1alpha1.OptInRequestTodo)
+	if currentInFlightCount <= MaxOptInRegionRequest {
+		reqLogger.Info(fmt.Sprintf("currentInFlightCount (%d) <= maxOpenOptInRegionRequests (%d)", currentInFlightCount, MaxOptInRegionRequest))
+		for region, onlyOpenOptInRequest := range onlyOpenOptInRequests {
+			if _, ok := inFlightOptInRequests[region]; !ok {
+				inFlightOptInRequests[region] = &awsv1alpha1.OptInRegionStatus{}
+			}
+			inFlightOptInRequests[region] = onlyOpenOptInRequest
+			currentInFlightCount += 1
+			if currentInFlightCount >= MaxOptInRegionRequest {
+				break
+			}
+		}
+	}
+	reqLogger.Info("Handling region request", "current-in-flight-count", currentInFlightCount)
+	err := updateOptInRegionRequests(reqLogger, awsClientBuilder, awsSetupClient, currentAcctInstance, client, inFlightOptInRequests, currentInFlightCount)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	err = client.Status().Update(context.TODO(), currentAcctInstance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	return reconcile.Result{RequeueAfter: 60 * time.Second, Requeue: true}, err
+}
+
+func updateOptInRegionRequests(reqLogger logr.Logger, awsClientBuilder awsclient.IBuilder, awsSetupClient awsclient.Client, currentAcctInstance *awsv1alpha1.Account, client client.Client, optInRequests awsv1alpha1.OptInRegions, count int) error {
+	for region, regionRequest := range optInRequests {
+		regionLogger := reqLogger.WithValues("Region", region)
+		roleToAssume := currentAcctInstance.GetAssumeRole()
+		awsAssumedRoleClient, _, err := AssumeRoleAndCreateClient(reqLogger, awsClientBuilder, currentAcctInstance, client, awsSetupClient, region, roleToAssume, "")
+		if err != nil {
+			reqLogger.Error(err, "Could not impersonate AWS account", "aws-account", currentAcctInstance.Spec.AwsAccountID)
+			return err
+		}
+		reqLogger.Info(fmt.Sprintf("Handling Opt-In region request for %s", region))
+		err = HandleOptInRegionRequests(regionLogger, awsAssumedRoleClient, region, regionRequest, currentAcctInstance)
+		if err != nil {
+			return err
+		}
+
+	}
+	return nil
+}
+
+func enableOptInRegions(reqLogger logr.Logger, client awsclient.Client, regionCode string) (bool, error) {
+	var result *account.EnableRegionOutput
+	var alreadySubmitted bool
+
+	// Default is 1/10 of a second, but any retries we need to make should be delayed a few seconds
+	// This also defaults to an exponential backoff, so we only need to try ~5 times, default is 10
+	retry.DefaultDelay = 3 * time.Second
+	retry.DefaultAttempts = uint(5)
+	err := retry.Do(
+		func() (err error) {
+			result, err = client.EnableRegion(&account.EnableRegionInput{
+				RegionName: aws.String(regionCode),
+			})
+			if err != nil {
+				if aerr, ok := err.(awserr.Error); ok {
```

Comment:
Can we replace this with an `errors.As` ? I know we use this structure elsewhere in the codebase but it can lead to mistakes with wrapped errors. `errors.As` traverses all of the wrapped errors while checking for a specific error type.

There are also a few other places that do this in this file as well. I'll leave it to your discretion.